picamera2/request.py

```diff
@@ -269,7 +277,10 @@ def save(self, img, metadata, file_output, format=None):
                 exif_ifd = {piexif.ExifIFD.DateTimeOriginal: datetime_now,
                             piexif.ExifIFD.ExposureTime: (metadata["ExposureTime"], 1000000),
                             piexif.ExifIFD.ISOSpeedRatings: int(total_gain * 100)}
-                exif = piexif.dump({"0th": zero_ifd, "Exif": exif_ifd})
+                exif_dict = {"0th": zero_ifd, "Exif": exif_ifd}
+                # merge user provided exif data, overwriting the defaults
+                exif_dict = exif_dict | exif_data
```

Comment:
This line works in Python 3.9+, for Python 3.5+ instead, we would do: `exif_dict = {**exif_dict, **exif_data}`, but I wasn't sure which Python version this is based on...